### PR TITLE
Upgrade to golangci-lint 1.31.0

### PR DESCRIPTION
**Pull request by cben**
- I believe most of us have an old golangci-lint 1.10.2 in our PATH, to match what our internal Jenkins has for internal OCM projects.  
  Running from container will allow each repo to control golangci-lint version independently.

- ~~Added `make check` for everything worth running before PR submission.
  Poor man's CI (#67) :grin:~~

If we like these, I can make similar PRs to sdk, moactl etc...

### CODE REVIEW

**Comment by cben** [comment]
oh, I didn't realize we already have a pr_check.sh wired to CI :tada:
debugging the failure...

**Comment by cben** [comment]
This is ready.

File: Makefile [hunk 1]
```diff
@@ -21,6 +21,9 @@ export GOPROXY=https://proxy.golang.org
 # Disable CGO so that we always generate static binaries:
 export CGO_ENABLED=0
 
+# Allow overriding: `make lint container_runner=podman`.
+container_runner:=docker
```
Comment (igoihman) [inline Makefile:25]: can we use podman instead?

Comment (cben) [reply Makefile:25]: It gave some problem, probably too old in ci-ext Jenkins. Let me try again...

**Comment by cben** [comment]
Traced this to https://access.redhat.com/solutions/5106161, ci.ext has has kernel 3.10.0-1062.9.1.el7.x86_64.  Asked app-sre if it's possible to upgrade to kernel-3.10.0-1127.8.2.el7.x86_64.

But meanwhile reverted to docker, podman won't work in CI.

I you want, I can set the Makefile default to podman and override to docker in pr_check.sh.

**Comment by cben** [comment]
Kernel upgraded today, now defaults to podman and works :tada:.
